### DISCUSSION TITLE
Cannot read properties of undefined (reading 'length') when using useSession()

### DISCUSSION THREAD

**Discussion by pepew-le-boss** [nextauthjs/next-auth#8522] [Help]
### Environment

next: 13.4.19
next-auth: 4.23.1
react: 18.2.0

### Reproduction URL

N/A

### Describe the issue

In my NextJS 13 project I've set up my auth and everything was working fine. 
I upgraded my `next` version from `13.4.4` to `13.4.19` and `next-auth` from `4.22.4` to `4.23.1`.

Now, when I start my app I get the following error: _Cannot read properties of undefined (reading 'length')_. Pointing at the `useSession()` hook.

So basically everywhere I have a client component with a `useSession()`, I have this error.

For example the first place getting triggered is my session error handler used in my `RootLayout`.
Here's the component: 
```javascript
"use client"

import { DialogError } from "@/components/client/common/DialogError"
import { signOut, useSession } from "next-auth/react"
import useTranslation from "next-translate/useTranslation"
import { useEffect, useState } from "react"

export const NextAuthErrorHandler = () => {
  const [openModal, setOpenModal] = useState(false)
  const [errorModal, setErrorModal] = useState("")

  const { t } = useTranslation("authentication")
  const { data: session } = useSession()

  useEffect(() => {
    async function displayErrorModal() {
      if (session?.error === "RefreshAccessTokenError") {
        setOpenModal(true)
        setErrorModal(t("error_refresh_token"))
        if (openModal === false) {
          // Quand on ferme la modal
          session.error = undefined
          await signOut({ callbackUrl: "/" })
        }
      }
    }
    displayErrorModal()
  }, [openModal, session, t])

  return <DialogError open={openModal} setOpen={setOpenModal} error={errorModal} />
}
```

But keep in mind that everywhere the `useSession()` is used I have this error. So if I comment the above code, the next `useSession()` in my code which is triggered raise the same error.

### How to reproduce

N/A

### Expected behavior

N/A

**Comment by pepew-le-boss** [comment]
Up pls
Still the same after bumping `next` to `13.5.2` and `next-auth` still at `4.23.1`

**Comment by alexandre-tobia** [comment]
Same error here.

**Comment by PI-YWT** [comment]
Is it resolved?

**Reply by pepew-le-boss** [reply]
No

**Comment by pepew-le-boss** [comment]
Still the same with `next` `14.0.2` and `next-auth` `4.24.4`

**Comment by pepew-le-boss** [comment]
After some investigation it's because of the `next-translate` package and more precisely the `useTranslation()` hook. I don't know why it's happening but I have the same error on some `useEffect()` or even `useState()` hooks.
It seems to happen only on components that can appear into view like modals or drawers but I might be wrong.

**Reply by alexandre-tobia** [reply]
After moving a component in another directory (between a layout and a page file) (don't remember why) the issue is gone on my side, it was very strange. Can't reproduce :/

**Comment by UmizDemud** [comment]
Not fixed huh? Same with useEffect hook.
Appeared right after I deleted the code which was giving another error:
```
[Unhandled Runtime Error: Should have a queue. This is likely a bug in React. Please file an issue](
```

Which was coming from trying to use a hook result and initialize a state variable with the hook's return value. Something exactly like:
```ts
const largeArrayOfObjects = useMyCustomHook();
const [myState, setMyState] = useState(largeArrayOfObjects);
```

Had to delete this way of implementation, which wasn't all too bad
